crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -255,6 +308,49 @@ impl<'a> LimitsTraversal<'a> {
         Ok(())
     }
 
+    /// multiGet queries can only pass a number of keys that does not exceed the max page size.
+    /// This checks for the number of keys and deducts the raw size of the keys from the payload
+    /// size.
+    fn check_multiget_arg(
```

Comment:
As multi-get functionality is achieved through connections, we do a pretty simple assumption that we will yield `connection_page_size` results, which is the `default_page_size` unless an override has been specified by the caller. I wonder if we could do something similar here, assume that a particular multi-get if invoked will yield the maximum allowed number of results?

Then we can do something similar, where in the resolver (such as `TransactionBlock::paginate`) we have the actual check for `ids.len() > limits.max_multiget -> return Error`

Reply:
How can I tell multi-get functionality is achieved through connections? 

This query does not return a page, but returns the number of keys passed in, which cannot be larger than max page size.

Reply:
Sorry for the confusion, what I meant is that today to do multi-get you go through `objects` connection, which assumes that we'll yield `default_page_size` results unless this is explicitly overwritten with `first` or `last`. Wonder if we could simplify by making a similar assumption, if we see `multiGet`, then we defer to the default or max limit for that `multiGet`

Reply:
I see. But this function still needs to account for those input object keys and remove them from the query limit. So I am not sure what you're thinking of.